app/controllers/macros_controller.rb

```diff
@@ -68,10 +72,6 @@ def command_params
     params.require(:commands).permit(ids: [])
   end
 
-  def find_project
-    @project = Project.find_by_param!(params[:project_id])
-  end
-
   def find_macro
     @macro = @project.macros.find(params[:id])
```

Comment:
What about if (assuming there is a method that does the same for project)
```ruby
def macro
 @macro ||= project.macros.find(params[:id])
end
```
and remove from the before action?